structuredheadings/plugin.js

```diff
@@ -305,10 +307,12 @@
           setLevel(editor, heading);
           setCurrentStyle(editor, heading, value);
         },
+        onClick: function (value) {  // eslint-disable-line max-statements
+          editor.fire("saveSnapshot");
```

Comment:
Why do we need to save a snapshot both before and after? That doesn't seem right. Isn't the pattern to save a snapshot after you do some work (which is what execCommand probably does automatically)?